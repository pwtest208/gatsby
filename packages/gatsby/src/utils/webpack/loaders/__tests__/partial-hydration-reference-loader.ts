import fs from "fs-extra"
import path from "path"

// @ts-ignore - Doesn't understand CJS default exports?
import partialHydrationReferenceLoader from "../partial-hydration-reference-loader"

const rootContext = `/` // TODO: Figure out what this should be
const fixturesPath = path.resolve(__dirname, `./fixtures`)
const fixtures = fs.readdirSync(fixturesPath)

for (const fixture of fixtures) {
  const resourcePath = path.resolve(__dirname, `./fixtures/${fixture}`)
  const source = fs.readFileSync(resourcePath).toString()
  const fixtureName = fixture.split(`.`)[0]?.replace(/-/g, ` `)

  it(`${fixtureName} scenario is handled`, async () => {
    const mutatedSource = await partialHydrationReferenceLoader.call(
      { resourcePath, rootContext },
      source
    )
    expect(mutatedSource).toMatchSnapshot()
  })
}
