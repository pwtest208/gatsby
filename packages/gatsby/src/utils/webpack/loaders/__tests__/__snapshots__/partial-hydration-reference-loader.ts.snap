// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`covers cases shown in fixtures/esm-aggregation.js 1`] = `""`;

exports[`covers cases shown in fixtures/esm-declaration.js 1`] = `
"export const a = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'a'
  }
export const b = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'b'
  }
export const c = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'c'
  }
export const d = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'd'
  }
export const e = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'e'
  }
export const f = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'f'
  }
export const functionName = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'functionName'
  }
export const ClassName = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'ClassName'
  }
export const g = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'g'
  }
export const foo = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'foo'
  }
export const i = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'i'
  }
export const j = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-declaration.js',
    name: 'j'
  }"
`;

exports[`covers cases shown in fixtures/esm-default-class-anonymous.js 1`] = `""`;

exports[`covers cases shown in fixtures/esm-default-class-named.js 1`] = `""`;

exports[`covers cases shown in fixtures/esm-default-expression.js 1`] = `""`;

exports[`covers cases shown in fixtures/esm-default-function-anonymous.js 1`] = `""`;

exports[`covers cases shown in fixtures/esm-default-function-named.js 1`] = `""`;

exports[`covers cases shown in fixtures/esm-list.js 1`] = `
"export const a = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-list.js',
    name: 'a'
  }
export const b = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-list.js',
    name: 'b'
  }
export const d = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-list.js',
    name: 'd'
  }
export const e = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-list.js',
    name: 'e'
  }
export const c = {
    $$typeof: Symbol.for('react.module.reference'),
    filepath: 'file:/<PROJECT_ROOT>/packages/gatsby/src/utils/webpack/loaders/__tests__/fixtures/esm-list.js',
    name: 'c'
  }"
`;

exports[`covers cases shown in fixtures/no-client-export.js 1`] = `""`;
